pkg/ingester/activeseries/model/custom_trackers_config.go

```diff
@@ -189,3 +189,39 @@ func NewCustomTrackersConfig(m map[string]string) (c CustomTrackersConfig, err e
 	c.string = customTrackersConfigString(c.source)
 	return c, nil
 }
+
+// MergeCustomTrackersConfig returns a new CustomTrackersConfig containing the merge of the two
+// CustomTrackersConfig in input. The two configs in input are not manipulated. If a key exists
+// in both configs, second config wins over first config.
+func MergeCustomTrackersConfig(first, second CustomTrackersConfig) CustomTrackersConfig {
+	if len(first.config) == 0 && len(second.config) == 0 {
+		return CustomTrackersConfig{}
+	}
```

Comment:
```suggestion
	if len(first.config) == 0 && len(second.config) == 0 {
		return CustomTrackersConfig{}
	}
	if len(second.config) == 0 {
		return first
	}
	if len(first.config) == 0 {
		return second
	}
```

Reply:
Good point. Addressed in https://github.com/grafana/mimir/pull/10428/commits/76bdd959ebfa6f6263f3cd4511b7f41a7137d01d.